core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1211,16 +1285,10 @@ write_files:
                         break
                       fi
 
-                      if [ $((secs % SG_POLL_INTERVAL)) -eq 0 ]; then
-                        # Reset counter to avoid overflow
-                        secs=0
-
-                        # AutoScalingGroup lifecycle check
-                        state=$(asg describe-auto-scaling-instances --region "${REGION}" --instance-ids "${INSTANCE_ID}" | jq -r '.AutoScalingInstances[].LifecycleState')
-                        if [ "${state}" = Terminating:Wait ]; then
-                          termination_source=asg
-                          break
-                        fi
+                      # Termination annotation check
+                      termination_source=$(kubectl get nodes "${NODE_NAME}" -o json | jq -r '.metadata.annotations["kube-aws.coreos.com/node-termination-source"]')
+                      if [ "${termination_source}" == asg ]; then
+                        break
                       fi
                     done
 
```

Comment:
If the `complete-lifecycle-action` command fails, the only side effect is that we'd have to wait for the drain timeout to complete before proceeding. Also, the loop will restart after 300s, where it will try to evict the node again (this is basically a no-op since all evictable pods were already moved to other nodes at this point), but yes, I agree that we could wrap the `complete-life-cycle` command in its own loop to avoid waiting too long before proceeding in case the command fails the first time.